src/brackets.js

```diff
@@ -142,7 +142,42 @@ define(function (require, exports, module) {
     
     // read URL params
     params.parse();
-    
+
+    // Define a preference for font smoothing mode on Mac.
+    // By default fontSmoothing is set to subpixel-antialiased
+    // for the text inside code editor. It can be overridden
+    // to "antialiased", that would set text rendering AA to use
+    // gray scale antialiasing.
+
+    if (brackets.platform === "mac") {
+
+        PreferencesManager.definePreference("fontSmoothing", "string", "subpixel_antialiased", {
+            description: Strings.DESCRIPTION_FONT_SMOOTHING,
+            values: ["subpixel-antialiased", "antialiased"]
+        });
+
+        PreferencesManager.on("change", "fontSmoothing", function () {
+
+            var aaType = PreferencesManager.get("fontSmoothing");
+
+            // For any other value, set this to subpixel-antialiased
+            if (aaType !== "subpixel-antialiased" && aaType !== "antialiased") {
```

Comment:
Taking another look at this line, I think we could also make it a Boolean pref, as we only really have two values (and I don't think there will be more in the future)